alacritty_terminal/src/term/mod.rs

```diff
@@ -614,18 +618,20 @@ fn cubic_bezier(p0: f64, p1: f64, p2: f64, p3: f64, x: f64) -> f64 {
         + x.powi(3) * p3
 }
 
-impl VisualBell {
-    pub fn new(config: &Config) -> VisualBell {
-        let visual_bell_config = &config.visual_bell;
-        VisualBell {
-            animation: visual_bell_config.animation,
-            duration: visual_bell_config.duration(),
+impl Bell {
+    pub fn new(config: &Config) -> Bell {
+        let bell_config = &config.bell;
+        Bell {
+            animation: bell_config.animation,
+            duration: bell_config.duration(),
+            command: bell_config.command.clone(),
             start_time: None,
         }
     }
 
     /// Ring the visual bell, and return its intensity.
     pub fn ring(&mut self) -> f64 {
+        self.run_command();
```

Comment:
I moved the running of the command to the start of the `ring()` method, instead of calling it separate from that...